### PR TITLE
Transaction address

### PR DESCRIPTION
**Pull request by jonano614**
It is easier now to track your transaction.

default

Also some renaming were done.

### CODE REVIEW

File: client/commands.c [hunk 1]
```diff
@@ -28,12 +28,13 @@ struct xfer_callback_data {
 	struct xdag_field fields[XFER_MAX_IN + 1];
 	int keys[XFER_MAX_IN + 1];
 	xdag_amount_t todo, done, remains;
-	int nfields, nkeys, outsig;
+	int fieldsCount, keysCount, outsig;
+	xdag_hash_t transactionBlockHash;
```
Comment (jonano614) [inline client/commands.c]: New field "transactionBlockHash".

File: client/commands.c [hunk 2]
```diff
@@ -459,50 +460,59 @@ int xdag_do_xfer(void *outv, const char *amount, const char *address, int isGui)
 	g_xdag_state = XDAG_STATE_XFER;
 	g_xdag_xfer_last = time(0);
 	xdag_traverse_our_blocks(&xfer, &xfer_callback);
-	if (out) {
-		fprintf(out, "Xfer: transferred %.9Lf %s to the address %s, see log for details.\n",
+	if(out) {
+		fprintf(out, "Xfer: transferred %.9Lf %s to the address %s.\n",
 			amount2xdags(xfer.done), g_coinname, xdag_hash2address(xfer.fields[XFER_MAX_IN].hash));
+		fprintf(out, "Transaction address is %s, it will take several minutes to complete the transaction.\n",
```
Comment (jonano614) [inline client/commands.c]: New message for transaction.

File: client/block.c [hunk 3]
```diff
@@ -729,15 +729,17 @@ int xdag_add_block(struct xdag_block *b)
  * in the following 'noutput' fields similarly - outputs, fee; send_time (time of sending the block);
  * if it is greater than the current one, then the mining is performed to generate the most optimal hash
  */
-int xdag_create_block(struct xdag_field *fields, int ninput, int noutput, xdag_amount_t fee, xdag_time_t send_time)
+int xdag_create_block(struct xdag_field *fields, int inputsCount, int outputsCount, xdag_amount_t fee, 
+    xdag_time_t send_time, xdag_hash_t newBlockHashResult)
```
Comment (jonano614) [inline client/block.c]: New parameter "newBlockHashResult" is added to xdag_create_block function.

File: client/block.c [hunk 4]
```diff
@@ -851,25 +856,30 @@ int xdag_create_block(struct xdag_field *fields, int ninput, int noutput, xdag_a
...
+                newBlockHash, sizeof(xdag_hash_t));
 			memcpy(g_xdag_mined_nonce[MAIN_TIME(send_time) & (XDAG_POOL_N_CONFIRMATIONS - 1)],
-				   b[0].field[XDAG_BLOCK_FIELDS - 1].data, sizeof(xdag_hash_t));
+                block[0].field[XDAG_BLOCK_FIELDS - 1].data, sizeof(xdag_hash_t));
 		}
 
-		xdag_send_new_block(b); res = 0;
+		xdag_send_new_block(block);
+
+        if(newBlockHashResult != NULL) {
```
Comment (jonano614) [inline client/block.c]: Hash of a new block is copied to result buffer in case of success.

**Comment by xrdavies** [comment]
Thank you for your efforts!

**Pull request merged**